cmake/ActsFunctions.cmake

```diff
@@ -0,0 +1,78 @@
+# This function adds a helper target called ${target}_HEADERS which has
+# generated source files that include each hedaer that was given one-by-one.
+# The generated target links against the one given in `target` and therefore
+# has access to all includes.
+# The generated target is not included in the default build, and is only meant
+# to help tools like clangd discover compiler flags.
+function(acts_compile_headers target)
+    set(options "")
+    set(oneValueArgs GLOB)
```

Comment:
Maybe I missed the section where `GLOB` was not used as an argument. But if in fact `GLOB` is always used, couldn't we get rid of quite a bit of argument parsing code here?

Reply:
The argument parsing is largely done by `cmake_parse_args`, and I'd keep using this to have the keyword `GLOB` in front of the actual glob pattern. The other argument to my mind doesn't add much complexity, but I can remove it and always require the glob if you prefer.

Reply:
Hmm so if I would write the code, I would probably strip `GLOB` and the other argument away, but I'm good also with just removing the second argument. If needed, this can be re-added in the futuer

Reply:
removed the other arguments and kept only GLOB.